### ISSUE TITLE
Selectively enable back OR forward navigation gestures?

### ISSUE THREAD

**Issue by lougreenwood** [apache/cordova-plugin-wkwebview-engine#51]
There are scenarios where pages may want to allow a forward navigation gesture whilst disabling back, for example, in an app with multiple top-level routes:

```
- Top-level-route-1 (forward only)
  - Sub route 1 (back or forward)
    - Sub route 2 (back or forward)
- Top-level-route-2 (forward only)
  - Sub route 3 (back or forward)
    - Sub route 4 (back or forward)
```

In an app with multiple top-level routes, the current behaviour allows swiping back to navigate between top level routes - IMO this breaks the logical organisation of routes in the application structure where top-level routes are independent. additionally, I believe that this could cause confusion for users as it breaks the usual iOS app expectations.

tldr; The current `AllowBackForwardNavigationGestures` creates a behaviour which is more like a website than an app. Improving the configuration to enable forward whilst disabling back (or vice versa) allows for more 'app-like' navigation and is a step away from the uncanny valley of Cordova apps.

**Comment by janpio** [comment]
Isn't this something that can be customized in the web app?

**Comment by lougreenwood** [comment]
I don't see how, isn't this a WKWebView-level feature?

Sure, I can selectively enable/disable the swipe navigation with `window.WkWebView.allowsBackForwardNavigationGestures()` - but this enables both back and forward.

And since it seems that `allowsBackForwardNavigationGestures` simply navigates the WKWebView history, navigating between multiple top-level routes has a history and back swiping is allowed.

For example, if you open the Apple Music app:

```
- Library
  - Playlists
    - Some playlist
```

You can navigate down the route path, and you can swipe back up it, but once you reach the Library page, you can't swipe to go back to previous top-level route like Browse.

(Additionally, something like Apple Music doesn't appear to allow forward swiping at all, something this feature request would also allow us to emulate).

**Comment by janpio** [comment]
Are we talking about a Single Page App (e.g. Ionic) or about classic HTML pages that link to each other?

**Comment by lougreenwood** [comment]
SPA

**Comment by lougreenwood** [comment]
I just created a separate issue (for the benefit of future visitors from Google) because I'm now beginning to believe that I'm not using the feature as it's intended...

**Comment by 349989153** [comment]
+1

**Comment by 349989153** [comment]
@lougreenwood Have you found the solution?

**Comment by lougreenwood** [comment]
Nope - still in the uncanny valley :(

**Comment by lougreenwood** [comment]
@janpio do you have any other thoughts? Seems that the support label is possibly incorrect?

**Comment by janpio** [comment]
No thoughts. 

> Seems that the support label is possibly incorrect?

Why? We have 3 categories here: Bug reports, Feature requests and Support questions. I consider this a support question where a user is looking for individual help to solve one of their problems.

**Comment by lougreenwood** [comment]
Seems to me this is actually a feature request, from the initial ticket:

>tldr; The current AllowBackForwardNavigationGestures creates a behaviour which is more like a website than an app. Improving the configuration to enable forward whilst disabling back (or vice versa) allows for more 'app-like' navigation and is a step away from the uncanny valley of Cordova apps.

**Comment by jcesarmobile** [comment]
as explained in https://github.com/apache/cordova-plugin-wkwebview-engine/issues/52#issuecomment-525862460, this just allows to set/unset a wkwebview property, and the property allows both forward and backward, it doesn't offer a way of only forward or only backward
